### PR TITLE
mail-client/geary: bump 3.34.1

### PR DESCRIPTION
**Pull request by okias**
- added dep on gspell[vala]
 - added optional dep on net-mail/ytnef
 - added dep on gui-libs/libhandy
 - bumped to EAPI 7
 - raised vala dep to 0.44
 - implement BDEPEND
    
 Compile, run and usage tested.
    
 Closes: https://bugs.gentoo.org/697050

### CODE REVIEW

**Comment by okias** [comment]
Depends on: https://github.com/gentoo/gentoo/pull/13072

File: mail-client/geary/geary-3.34.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+VALA_MIN_API_VERSION="0.40"
```
Comment (cjmayo) [inline mail-client/geary/geary-3.34.1.ebuild]: I think this needs to be VALA_MIN_API_VERSION="0.44"

Build fails with dev-lang/vala-0.42.7:
```
../geary-3.34.1/src/client/components/client-web-view.vala:310.9-310.38: error: The name `enable_javascript_markup' does not exist in the context of `WebKit.Settings?'
        setts.enable_javascript_markup = false;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
enable_javascript_markup needs webkit 2.24 support in Vala
https://github.com/GNOME/vala/blob/ff57dee633975cba39883af7caf4d21e4909402e/vapi/webkit2gtk-4.0.vapi#L653
First added in Vala 0.44.0
https://github.com/GNOME/vala/commit/8a2e9e0f01b82bd81ab9ddf895733c6bf551fccc
https://github.com/GNOME/vala/blob/f413cb32470d9c2598416772bc0166c6f974ec9f/NEWS#L305

File: mail-client/geary/geary-3.34.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	>=gui-libs/libhandy-0.0.9:0.0=
+	>=x11-libs/gtk+-3.22.26:3[introspection]
+	>=net-libs/webkit-gtk-2.20:4=[introspection]
+	app-text/gspell[vala]
+	app-text/iso-codes
+	>=dev-db/sqlite-3.12:3
+
+	>=net-libs/libsoup-2.48:2.4[introspection]
+	net-libs/gnome-online-accounts
+	>=net-mail/ytnef-1.9.3
```
Comment (cjmayo) [inline mail-client/geary/geary-3.34.1.ebuild]: net-mail/ytnef can be made optional using `-Dtnef-support` in emesonargs (tested with geary-3.34.0).

Comment (okias) [reply mail-client/geary/geary-3.34.1.ebuild]: do you have somewhere ebuild with this change?

File: mail-client/geary/geary-3.34.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+RDEPEND="${DEPEND}
+	gnome-base/dconf
+	gnome-base/gsettings-desktop-schemas
+"
+# org.gnome.desktop.interface clock-format global setting usage
+# gnome-doc-utils for xml2po for TRANSLATE_HELP option
+DEPEND="${DEPEND}
+	sys-devel/gettext
+	dev-util/desktop-file-utils
+	$(vala_depend)
```
Comment (mattst88) [inline mail-client/geary/geary-3.34.2.ebuild]: Probably should go in `BDEPEND`

File: mail-client/geary/geary-3.34.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+"
+
+RDEPEND="${DEPEND}
+	gnome-base/dconf
+	gnome-base/gsettings-desktop-schemas
+"
+# org.gnome.desktop.interface clock-format global setting usage
+# gnome-doc-utils for xml2po for TRANSLATE_HELP option
+DEPEND="${DEPEND}
+	sys-devel/gettext
```
Comment (mattst88) [inline mail-client/geary/geary-3.34.2.ebuild]: Probably should go in `BDEPEND`

File: mail-client/geary/geary-3.34.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	dev-util/desktop-file-utils
```
Comment (mattst88) [inline mail-client/geary/geary-3.34.2.ebuild]: Not sure. Maybe in `BDEPEND`? Looks like it only installs binaries, man pages, etc.

File: mail-client/geary/geary-3.34.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+BDEPEND="
+	virtual/pkgconfig
+"
+
+RDEPEND="${DEPEND}
+	gnome-base/dconf
+	gnome-base/gsettings-desktop-schemas
+"
+# org.gnome.desktop.interface clock-format global setting usage
+# gnome-doc-utils for xml2po for TRANSLATE_HELP option
```
Comment (mattst88) [inline mail-client/geary/geary-3.34.2.ebuild]: No `gnome-doc-utils` dependency, so this comment can be removed.

**Pull request closed without merging**